onnxconverter_common/onnx_ops.py

```diff
@@ -423,19 +423,31 @@ def apply_normalization(scope, input_name, output_name, container, operator_name
 def apply_pad(scope, input_name, output_name, container, operator_name=None, mode=None, pads=None, value=None):
     name = _create_name_or_use_existing_one(scope, 'Pad', operator_name)
     attrs = {'name': name}
+    inputs = [input_name]
 
     if mode is not None:
         attrs['mode'] = mode
-    if value is not None:
-        attrs['value'] = value
-    if container.target_opset < 2:
-        attrs['paddings'] = pads
-        op_version = 1
-    else:
-        attrs['pads'] = pads
-        op_version = 2
 
-    container.add_node('Pad', input_name, output_name, op_version=op_version, **attrs)
+    if container.target_opset < 11:
+        if value is not None:
+            attrs['value'] = value
+        if container.target_opset < 2:
+            attrs['paddings'] = pads
+            op_version = 1
+        else:
+            attrs['pads'] = pads
+            op_version = 2
+    else:
+        op_version = 11
+        pads_name = scope.get_unique_variable_name(name + '_pads')
+        container.add_initializer(pads_name, onnx_proto.TensorProto.INT64, [len(pads)], pads)
```

Comment:
Also support pads being a dynamic input?

Reply:
The pads dimension equals to input dimension, there is no dynamic there.